deploy-service/common/src/main/java/com/pinterest/deployservice/db/DBHostAgentDAOImpl.java

```diff
@@ -15,30 +15,40 @@
 
-import com.pinterest.deployservice.bean.HostAgentBean;
-import com.pinterest.deployservice.bean.SetClause;
-import com.pinterest.deployservice.dao.HostAgentDAO;
-
 public class DBHostAgentDAOImpl implements HostAgentDAO {
-    private static final String INSERT_HOST_TEMPLATE = "INSERT INTO hosts_and_agents SET %s ON DUPLICATE KEY UPDATE %s";
-    private static final String UPDATE_HOST_BY_ID = "UPDATE hosts_and_agents SET %s WHERE host_id=?";
+    private static final String INSERT_HOST_TEMPLATE =
+            "INSERT INTO hosts_and_agents SET %s ON DUPLICATE KEY UPDATE %s";
+    private static final String UPDATE_HOST_BY_ID =
+            "UPDATE hosts_and_agents SET %s WHERE host_id=?";
     private static final String DELETE_HOST_BY_ID = "DELETE FROM hosts_and_agents WHERE host_id=?";
-    private static final String GET_HOST_BY_NAME = "SELECT * FROM hosts_and_agents WHERE host_name=?";
-    private static final String GET_HOST_BY_HOSTID = "SELECT * FROM hosts_and_agents WHERE host_id=?";
-    private static final String GET_HOSTS_BY_LAST_UPDATE = "SELECT DISTINCT * FROM hosts_and_agents WHERE last_update<?";
-    private static final String GET_HOSTS_BY_LAST_UPDATES = "SELECT DISTINCT * FROM hosts_and_agents WHERE last_update>? AND last_update<?";
-    private static final String GET_STALE_ENV_HOST = "SELECT DISTINCT hosts_and_agents.* FROM hosts_and_agents INNER JOIN hosts_and_envs ON hosts_and_agents.host_name=hosts_and_envs.host_name WHERE hosts_and_agents.last_update<?";
-    private static final String GET_HOSTS_BY_AGENT = "SELECT * FROM hosts_statuses WHERE agent_version=? ORDER BY host_id LIMIT ?,?";
-    private static final String GET_DISTINCT_HOSTS_COUNT = "SELECT COUNT(DISTINCT host_id) FROM hosts_and_agents";
+    private static final String GET_HOST_BY_NAME =
+            "SELECT * FROM hosts_and_agents WHERE host_name=?";
+    private static final String GET_HOST_BY_HOSTID =
+            "SELECT * FROM hosts_and_agents WHERE host_id=?";
+    private static final String GET_HOSTS_BY_LAST_UPDATE =
+            "SELECT DISTINCT * FROM hosts_and_agents WHERE last_update<?";
+    private static final String GET_HOSTS_BY_LAST_UPDATES =
+            "SELECT DISTINCT * FROM hosts_and_agents WHERE last_update>? AND last_update<?";
+    private static final String GET_STALE_ENV_HOST =
+            "SELECT DISTINCT hosts_and_agents.* FROM hosts_and_agents INNER JOIN hosts_and_envs ON hosts_and_agents.host_name=hosts_and_envs.host_name WHERE hosts_and_agents.last_update<?";
+    private static final String GET_HOSTS_BY_AGENT =
+            "SELECT * FROM hosts_statuses WHERE agent_version=? ORDER BY host_id LIMIT ?,?";
+    private static final String GET_DISTINCT_HOSTS_COUNT =
+            "SELECT COUNT(DISTINCT host_id) FROM hosts_and_agents";
+    private static final String GET_MAIN_ENV_BY_HOSTID =
+            "SELECT e.* FROM hosts_and_agents ha JOIN environs e ON ha.auto_scaling_group = e.cluster_name WHERE ha.host_id = ?";
```

Comment:
How does it work with sidecar owner? Would a sidecar owner still be able to restart the sidecar even though they don't own the cluster or does it somehow fall back to sidecar owner permissions?

Reply:
That's a good question, I was gonna update the description to elaborate the new limitation.